src/software/utils/main_sfmFilter.cpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2024 AliceVision contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <aliceVision/sfmData/SfMData.hpp>
+#include <aliceVision/sfmDataIO/sfmDataIO.hpp>
+#include <aliceVision/sfm/utils/alignment.hpp>
+
+#include <aliceVision/cmdline/cmdline.hpp>
+#include <aliceVision/system/main.hpp>
+#include <aliceVision/system/Logger.hpp>
+#include <aliceVision/stl/regex.hpp>
+
+#include <boost/program_options.hpp>
+
+#include <string>
+#include <sstream>
+
+// These constants define the current software version.
+// They must be updated when the command line is changed.
+#define ALICEVISION_SOFTWARE_VERSION_MAJOR 1
+#define ALICEVISION_SOFTWARE_VERSION_MINOR 0
+
+using namespace aliceVision;
+using namespace aliceVision::sfm;
+
+namespace po = boost::program_options;
+
+int aliceVision_main(int argc, char** argv)
+{
+    system::Timer timer;
+
+    // command-line parameters
+    std::string sfmDataFilename;
+    std::string outputSfMFilenameSelected;
+    std::string outputSfMFilenameUnselected;
+    std::string fileMatchingPattern;
+
+    // clang-format off
+    po::options_description requiredParams("Required parameters");
+    requiredParams.add_options()
+        ("inputFile,i", po::value<std::string>(&sfmDataFilename)->required(),
+         "Path to the input SfMData file.\n")
+        ("fileMatchingPattern,m", po::value<std::string>(&fileMatchingPattern)->required(),
+         "Matching pattern for the from_filepath method.\n")
+        ("outputSfMData_selected,o", po::value<std::string>(&outputSfMFilenameSelected)->required(),
+         "Path to the output SfMData file.\n")
+         ("outputSfMData_unselected,o", po::value<std::string>(&outputSfMFilenameUnselected)->required(),
+         "Path to the output SfMData file.\n");
+
```

Comment:
```suggestion
    // clang-format on
    
```

clang-format is disabled at the beginning of the parameters listing but it is never re-enabled. It should be re-enabled as soon as the parameters have been described.